lib/state_change_syncer.go

```diff
@@ -455,6 +460,16 @@ func (stateChangeSyncer *StateChangeSyncer) _handleStateSyncerOperation(event *S
 		encoderType = keyEncoder.GetEncoderType()
 		stateChangeEntry.Encoder = keyEncoder
 		stateChangeEntry.EncoderBytes = nil
+
+		if stateChangeEntry.AncestralRecordBytes != nil && len(stateChangeEntry.AncestralRecordBytes) > 0 {
+			// Decode the ancestral record.
+			ancestralRecord, err := DecodeStateKey(stateChangeEntry.KeyBytes, stateChangeEntry.AncestralRecordBytes)
+			if err != nil {
+				glog.Fatalf("Server._handleStateSyncerOperation: Error decoding ancestral record: %v", err)
```

Comment:
Fatal is going to kill the node if we hit this error. Are we sure that's what we want?

Reply:
This function is unfortunately not in a clean place for errors to get passed up the chain and handled. That said, at the top of the chain here we would probably be throwing a fatal one if this were to trip (it never has in all the time we've been running this), since this does mean that our entire state chain is somehow messed up. This also should have no impact on non state-sync nodes.

Reply:
Makes sense. Sounds good to me!